devito/passes/clusters/aliases.py

```diff
@@ -849,7 +849,8 @@ def lower_schedule(schedule, meta, sregistry, ftemps):
         # This prevents cases such as `floor(a*b)` with `a` and `b` floats
         # that would creat a temporary `int r = b` leading to erronous
         # numerical results
-        dtype = sympy_dtype(pivot, base=meta.dtype)
+        mindtype = None if writeto else mindtype
```

Comment:
nitpicking, could be `min_dtype`, otherwise I'm tempted to read it (for the first 1-2 secs) as "mind_type"

Reply:
also, we tend not to have inline conditionals around

Reply:
in fact, what you should do instead is the following:

1) move `dtype = sympy_dtype(...)` inside the `if writeto:` below; in this case, don't use any `min_dtype`
2) also replicate `dtype = sympy_dtype(...)` in the corresponding else, but this time use the `min_dtype`

this way it gets way more explicit and clear